### PR TITLE
Issue 48 - Separate default and agent containers

### PR DESCRIPTION
**Pull request by cthiel1**
Integration Tests:
https://jenkins-jenkins.apps.tssc.rht-set.com/blue/organizations/jenkins/Ploigos%20Reference%20Applciations%20(everything)%2Freference-quarkus-mvn/detail/PR-19/2/pipeline
https://jenkins-jenkins.apps.tssc.rht-set.com/job/Ploigos%20Reference%20Applciations%20%28typical%29/job/reference-quarkus-mvn/job/PR-19/2/display/redirect
https://jenkins-jenkins.apps.tssc.rht-set.com/job/Ploigos%20Reference%20Applciations%20%28everything%29/job/reference-quarkus-mvn/job/PR-19/2/display/redirect

### CODE REVIEW

File: vars/ploigosWorkflowEverything.groovy [hunk 1]
```diff
@@ -124,7 +124,12 @@ class WorkflowParams implements Serializable {
     /* Container image to use when creating a workflow worker
      * to run pipeline steps when no other specific container image has been
      * specified for that step. */
-    String workflowWorkerImageDefault = "ploigos/ploigos-ci-agent-jenkins:latest"
+    String workflowWorkerImageDefault = "ploigos/ploigos-base:latest"
+
+    /* Container image to use when creating a workflow worker
+     * to run pipeline steps when no other specific container image has been
+     * specified for that step. */
```
Comment (gfulton-redhat) [inline vars/ploigosWorkflowEverything.groovy]: Looks like you need to update the comment for the workflowWorkerImageAgent.

File: vars/ploigosWorkflowExistingContainerImageScan.groovy [hunk 2]
```diff
@@ -95,7 +95,12 @@ class WorkflowParams implements Serializable {
     /* Container image to use when creating a workflow worker
      * to run pipeline steps when no other specific container image has been
      * specified for that step. */
-    String workflowWorkerImageDefault = "ploigos/ploigos-ci-agent-jenkins:latest"
+    String workflowWorkerImageDefault = "ploigos/ploigos-base:latest"
+
+    /* Container image to use when creating a workflow worker
+     * to run pipeline steps when no other specific container image has been
+     * specified for that step. */
+    String workflowWorkerImageAgent = "ploigos/ploigos-ci-agent-jenkins:latest"
```
Comment (gfulton-redhat) [inline vars/ploigosWorkflowExistingContainerImageScan.groovy]: You may also need to update the comment b/c it is a duplicate of the `workflowWorkerImageDefault`.

File: vars/ploigosWorkflowMinimal.groovy [hunk 3]
```diff
@@ -124,7 +124,12 @@ class WorkflowParams implements Serializable {
     /* Container image to use when creating a workflow worker
      * to run pipeline steps when no other specific container image has been
      * specified for that step. */
-    String workflowWorkerImageDefault = "ploigos/ploigos-ci-agent-jenkins:latest"
+    String workflowWorkerImageDefault = "ploigos/ploigos-base:latest"
+
+    /* Container image to use when creating a workflow worker
+     * to run pipeline steps when no other specific container image has been
+     * specified for that step. */
+    String workflowWorkerImageAgent = "ploigos/ploigos-ci-agent-jenkins:latest"
```
Comment (gfulton-redhat) [inline vars/ploigosWorkflowMinimal.groovy]: Update comment.

**Review by gfulton-redhat — COMMENTED** [review]
You may want to try to DRY up the multiple declarations like `workflowWorkerImageDefault` and `workflowWorkerImageAgent` in a separate PR.

**Comment by cthiel1** [comment]
Update to comments committed per @gfulton-redhat review. Good catch, thanks!

**Review by itewk — APPROVED** [review]

**Comment by itewk** [comment]
assuming @gfulton-redhat is good with the change after @cthiel1 fixed the comments. merging.

**Pull request merged**